### ISSUE TITLE
Add logging to tell me what labels are wrong on model finetuning

### ISSUE THREAD

**Issue by koaning** [RasaHQ/rasa#7959]
**Description of Problem**:

I'm currently training/finetuning a model that has 150 labels. On fine-tuning, I seem to hit an issue.

Training NLU model... 2021-02-15 15:37:58 WARNING rasa.shared.utils.common - The Incremental Training feature is currently experimental and might change or be removed in the future 
 <g-emoji class="g-emoji" alias="microscope" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f52c.png">
 🔬 
 </g-emoji> Please share your feedback on it in the forum ([https://forum.rasa.com](https://forum.rasa.com)) to help us make this feature ready for production. Loading NLU model from models/nlu-20210215-120910.tar.gz for finetuning... 2021-02-15 15:37:59 INFO rasa.model - Data (nlu_labels) for finetune section changed. NLU model can not be fine-tuned.

It's telling me that the `nlu_labels` have change but I after inspecting this manually this does not seem to be the case. This may be a bug in Rasa, I do not know yet, but the error message needs to be improved. It would help a lot it would tell me **which** labels are the culprit.

**Solution**

This might be a better error message:

2021-02-15 15:37:59 INFO rasa.model - Data (nlu_labels) for finetune section changed. 2021-02-15 15:37:59 INFO rasa.model - Please check intent: `greet` 2021-02-15 15:37:59 INFO rasa.model - Please check intent: `goodbye`

**Comment by dakshvar22** [comment]
I think the reason why this was not added was because it is computationally expensive and also it would mean that we persist the set of labels along with the model. Currently this check works by comparing the persisted fingerprint of the training data labels with the new fingerprint of the training labels during fine-tuning.

**Comment by koaning** [comment]
We don't keep track of all the known intent labels in our saved model file? Scikit-Learn models typically store a `.classes_` property so it's known which classes exist. Is it really *that* expensive to pull/store this information?

**Comment by dakshvar22** [comment]
It's saved in the model file but this check happens much before the components are loaded. i'll let @joejuzl comment on what's the best way to incorporate it. I agree that including this info in log messages will help.

**Comment by joejuzl** [comment]
What @dakshvar22 says is correct. The way we track changes to training data and configs is by using fingerprint hashes. There are lots of different things that can change and that effect whether training is needed, or what kind of training can be done. This specific fingerprint looks at labels for intents, entities, entity_groups, entity_roles, and actions. It wouldn't be simple to redesign this area of the code. 
However I could envisage a separate component, or set of functionality that could compare training data or domain files to show diffs. Although, again, I think this would end up being quite complicated unless we did something "simple" like comparing whole files.
